src/parsers/timeseries_formats.jl

```diff
@@ -0,0 +1,234 @@
+end
+
+function get_value_columns(::Type{TimeseriesFormatDateTimePeriodAsColumn},
+                           file::CSV.File)
+    return [x for x in propertynames(file) if !in(x, (:DateTime, :Period))]
+end
+
+"""Return the column names with values."""
+function get_value_columns(::Type{TimeseriesFormatComponentsAsColumnsNoTime},
+                           file::CSV.File)
+    return propertynames(file)
+end
+
+"""Return the column names that specify the Period."""
+function get_period_columns(::Type{TimeseriesFormatPeriodAsColumn},
+                            file::CSV.File)
+    return [:Period]
+end
+
+function get_period_columns(::Type{TimeseriesFormatYMDPeriodAsHeader},
+                            file::CSV.File)
+    return [x for x in propertynames(file) if !in(x, (:Year, :Month, :Day))]
+end
+
+"""Return a sorted vector of unique timestamps specified in the CSV file."""
+function get_unique_timestamps(::Type{T}, file::CSV.File) where T <: TimeseriesFormat
+    timestamps = Set{Dates.DateTime}()
+
+    for i in 1:length(file)
+        push!(timestamps, get_timestamp(T, file, i))
+    end
+
+    return sort!(collect(timestamps))
+end
+
+"""Return a Dates.DateTime for the row in the CSV file."""
+function get_timestamp(::Type{TimeseriesFormatYMDPeriodAsColumn}, file::CSV.File,
+                       row_index::Int)
+    return Dates.DateTime(file.Year[row_index], file.Month[row_index], file.Day[row_index])
+end
+
+function get_timestamp(::Type{TimeseriesFormatDateTimePeriodAsColumn}, file::CSV.File,
+                       row_index::Int)
+    return Dates.DateTime(file.DateTime[row_index])
+end
+
+function get_timestamp(::Type{TimeseriesFormatYMDPeriodAsHeader}, file::CSV.File,
+                       row_index::Int)
+    return get_timestamp(TimeseriesFormatYMDPeriodAsColumn, file, row_index)
+end
+
+
+"""Return a TimeSeries.TimeArray representing the CSV file.
+
+This version of the function covers a special case where the timeseries data is only for
+one component and column header does not match that component's name. In other cases it
+assumes that the column headers match the names. component_name only needs to be passed in
+the first case.
+"""
+function read_time_array(
+                         ::Type{T},
+                         file::CSV.File,
+                         component_name=nothing;
+                         kwargs...
+                        ) where T <: TimeseriesFormatPeriodAsColumn
+    timestamps = Vector{Dates.DateTime}()
+    step = get_step_time(T, file, file.Period)
+
+    for i in 1:length(file)
+        timestamp = get_timestamp(T, file, i) + step * (i - 1)
+        push!(timestamps, timestamp)
+    end
+
+    value_columns = get_value_columns(T, file)
+    vals = [getproperty(file, x) for x in value_columns]
+
+    if length(value_columns) == 1 && !isnothing(component_name) && string(value_columns[1]) != component_name
+        # TODO DT: Hack. Do we really want to support this?
```

Comment:
I think we should remove this.  I think we should make it a requirement that the column name be a valid component name.

Reply:
Agreed

Reply:
This means that the 6 files matching data/RTS_GMLC/Reserves/.*regional_Spin_Up_R1.csv will have to be corrected or skipped.

Reply:
Done

Reply:
Removed from code